G1ANT.Addon.Net/SimplifiedMessageSummary.cs

```diff
@@ -1,27 +1,321 @@
+                    return messageSummary.Envelope.MessageId;
+                return string.Empty;
+            }
+        }
+
+        public UniqueId UniqueId
+        {
+            get
+            {
+                if (messageSummary != null)
+                    return messageSummary.UniqueId;
+                return UniqueId.Invalid;
+            }
+        }
+
+        public string Subject
+        {
+            get
+            {
+                if (fullMessage != null)
+                    return fullMessage.Subject;
+                else if (messageSummary != null)
+                    return messageSummary.NormalizedSubject;
+                return string.Empty;
+            }
+            set
+            {
+                if (fullMessage != null)
+                    fullMessage.Subject = value;
+                else if (messageSummary != null)
+                    messageSummary.Envelope.Subject = value;
+            }
+        }
+
+        public InternetAddressList To
+        {
+            get
+            {
+                if (fullMessage != null)
+                    return fullMessage.To;
+                else if (messageSummary != null)
+                    return messageSummary.Envelope.To;
+                return null;
+            }
+        }
+
+        public InternetAddressList From
+        {
+            get
+            {
+                if (fullMessage != null)
+                    return fullMessage.From;
+                else if (messageSummary != null)
+                    return messageSummary.Envelope.From;
+                return null;
+            }
+        }
+
+        public MailboxAddress Sender
+        {
+            set
+            {
+                if (fullMessage != null)
+                    fullMessage.Sender = value;
+                else
+                    throw new NotSupportedException("Cannot set Sender for recieved message");
+            }
+        }
+
+        public InternetAddressList Cc
+        {
+            get
+            {
+                if (fullMessage != null)
+                    return fullMessage.Cc;
+                else if (messageSummary != null)
+                    return messageSummary.Envelope.Cc;
+                return null;
+            }
+        }
+
+        public InternetAddressList Bcc
+        {
+            get
+            {
+                if (fullMessage != null)
+                    return fullMessage.Bcc;
+                else if (messageSummary != null)
+                    return messageSummary.Envelope.Bcc;
+                return null;
+            }
+        }
+
+        public bool IsReply
+        {
+            get
+            {
+                if (messageSummary != null)
+                    return messageSummary.IsReply;
+                return false;
+            }
+        }
+
+        public string Priority
+        {
+            get
+            {
+                if (fullMessage != null && fullMessage.Headers != null)
```

Comment:
I would say `if (fullMessage?.Headers != null)`, but if you rewrite this class into to separate ones a second part of existing condition will be enough.